extensions/amp-a4a/0.1/test/test-a4a-integration.js

```diff
@@ -0,0 +1,187 @@
+      expectRenderedInXDomainIframe(a4aElement, TEST_URL);
+    });
+  });
+
+  it('should fall back to 3p when the XHR fails', () => {
+    xhrMock.resetBehavior();
+    xhrMock.throws(new Error('Testing network error'));
+    // TODO(tdrl) Currently layoutCallback rejects, even though something *is*
+    // rendered.  This should be fixed in a refactor, and we should change this
+    // .catch to a .then.
+    return fixture.addElement(a4aElement).catch(error => {
+      expect(error.message).to.contain.string('Testing network error');
+      expect(error.message).to.contain.string('amp-a4a:');
+      expectRenderedInXDomainIframe(a4aElement, TEST_URL);
+    });
+  });
+
+  it('should fall back to 3p when extractCreative throws', () => {
+    sandbox.stub(MockA4AImpl.prototype, 'extractCreativeAndSignature').throws(
+        new Error('Testing extractCreativeAndSignature error'));
+    // TODO(tdrl) Currently layoutCallback rejects, even though something *is*
+    // rendered.  This should be fixed in a refactor, and we should change this
+    // .catch to a .then.
+    return fixture.addElement(a4aElement).catch(error => {
+      expect(error.message).to.contain.string(
+          'Testing extractCreativeAndSignature error');
+      expect(error.message).to.contain.string('amp-a4a:');
+      expectRenderedInXDomainIframe(a4aElement, TEST_URL);
+    });
+  });
+
+  it('should fall back to 3p when extractCreative returns empty sig', () => {
+    sandbox.stub(MockA4AImpl.prototype, 'extractCreativeAndSignature')
+        .onFirstCall().returns({
+          creative: utf8Encode(validCSSAmp.reserialized),
+          signature: null,
+        })
+        .onSecondCall().throws(new Error(
+            'Testing extractCreativeAndSignature should not occur error'));
+    return fixture.addElement(a4aElement).then(unusedElement => {
+      expectRenderedInXDomainIframe(a4aElement, TEST_URL);
+    });
+  });
+
+  it('should fall back to 3p when extractCreative returns empty creative',
+      () => {
+        sandbox.stub(MockA4AImpl.prototype, 'extractCreativeAndSignature')
+            .onFirstCall().returns({
+              creative: null,
+              signature: validCSSAmp.signature,
+            })
+            .onSecondCall().throws(new Error(
+            'Testing extractCreativeAndSignature should not occur error'));
+        // TODO(tdrl) Currently layoutCallback rejects, even though something
+        // *is* rendered.  This should be fixed in a refactor, and we should
+        // change this .catch to a .then.
+        return fixture.addElement(a4aElement).catch(error => {
+          expect(error.message).to.contain.string('Key failed to validate');
+          expect(error.message).to.contain.string('amp-a4a:');
+          expectRenderedInXDomainIframe(a4aElement, TEST_URL);
+        });
+      });
+});
+
+// TODO(@ampproject/a4a): Need a test that double-checks that thrown errors
```

Comment:
Done.  Though really this is a note to figure out how to integrate this into *all* tests.  Added some clarification to the TODO.